tests/Unit/Parallel/Test_ResourceInfo.cpp

```diff
@@ -119,16 +182,793 @@ void test_singleton_pack() {
+        resource_info.build_singleton_map(cache);
+      })(),
+      Catch::Contains(
+          "The total number of cores requested is less than or equal to the "
+          "number of cores that requested to be exclusive, i.e. without"));
+
+  CHECK_THROWS_WITH(
+      ([]() {
+        auto resource_info =
+            TestHelpers::test_option_tag<OptionTags::ResourceInfo<metavars>>(
+                "AvoidGlobalProc0: true\n"
+                "Singletons:\n"
+                "  FakeSingleton0:\n"
+                "    Proc: 0\n"
+                "    Exclusive: false\n");
+        [[maybe_unused]] const size_t proc =
+            resource_info.template proc_for<FakeSingleton<metavars, 0>>();
+      })(),
+      Catch::Contains("The singleton map has not been built yet. You must call "
+                      "build_singleton_map() before you call this function."));
+
+  CHECK_THROWS_WITH(
+      ([]() {
+        auto resource_info =
+            TestHelpers::test_option_tag<OptionTags::ResourceInfo<metavars>>(
+                "AvoidGlobalProc0: true\n"
+                "Singletons:\n"
+                "  FakeSingleton0:\n"
+                "    Proc: 0\n"
+                "    Exclusive: false\n");
+        [[maybe_unused]] const auto& procs_to_ignore =
+            resource_info.procs_to_ignore();
+      })(),
+      Catch::Contains("The singleton map has not been built yet. You must call "
+                      "build_singleton_map() before you call this function."));
+
+  CHECK_THROWS_WITH(
+      ([]() {
+        auto resource_info =
+            TestHelpers::test_option_tag<OptionTags::ResourceInfo<metavars>>(
+                "AvoidGlobalProc0: true\n"
+                "Singletons:\n"
+                "  FakeSingleton0:\n"
+                "    Proc: 0\n"
+                "    Exclusive: false\n");
+        [[maybe_unused]] const auto singleton_info =
+            resource_info.get_singleton_info<FakeSingleton<metavars, 0>>();
+      })(),
+      Catch::Contains("The singleton map has not been built yet. You must call "
+                      "build_singleton_map() before you call this function."));
+}
+
+template <typename Metavariables>
+Parallel::ResourceInfo<Metavariables> create_resource_info(
+    const bool avoid_global_proc_0,
+    const std::vector<std::pair<bool, int>>& singletons) {
+  std::string option_str =
+      "AvoidGlobalProc0: " + (avoid_global_proc_0 ? "true"s : "false"s) + "\n";
+  option_str += "Singletons:\n";
+  for (size_t i = 0; i < singletons.size(); i++) {
+    const bool exclusive = singletons[i].first;
+    const int proc = singletons[i].second;
+    option_str += "  FakeSingleton" + get_output(i) + ":\n";
+    option_str +=
+        "    Proc: " + (proc == -1 ? "Auto"s : get_output(proc)) + "\n";
+    option_str += "    Exclusive: " + (exclusive ? "true"s : "false"s) + "\n";
+  }
+
+  return TestHelpers::test_option_tag<OptionTags::ResourceInfo<Metavariables>>(
```

Comment:
add a `serialize_and_deserialize` here so you can simultaneously test all the serialization?